clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java

```diff
@@ -555,8 +557,23 @@ public synchronized void removeReporter(MetricsReporter reporter) {
 
     synchronized void registerMetric(KafkaMetric metric) {
         MetricName metricName = metric.metricName();
-        if (this.metrics.containsKey(metricName))
-            throw new IllegalArgumentException("A metric named '" + metricName + "' already exists, can't register another one.");
+
+        KafkaMetric existingMetric = this.metrics.get(metricName);
+        if (existingMetric != null) {
+            if (replaceOnDuplicate) {
+                log.error("The metric " + metricName + " is being replaced since it had already been registered. Please file a bug report.");
```

Comment:
Why not simply calling removeMetric()? Looks like it's doing almost same thing?

Reply:
I guess just to avoid doing a lookup in the map again, but that's harmless, and you're right the code is the same.

I'll do that.

Reply:
Done.